tutorials/animation/index.md

```diff
@@ -0,0 +1,874 @@
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> [[highlight]]with SingleTickerProviderStateMixin[[/highlight]] {
+  [[highlight]]Animation<double> animation;[[/highlight]]
+
+  [[highlight]]initState() {[[/highlight]]
+    [[highlight]]super.initState();[[/highlight]]
+    [[highlight]]AnimationController controller = new AnimationController([[/highlight]]
+        [[highlight]]duration: const Duration(milliseconds: 2000), vsync: this);[[/highlight]]
+    [[highlight]]animation = new Tween(begin: 0.0, end: 300.0).animate(controller)[[/highlight]]
+      [[highlight]]..addListener(() {[[/highlight]]
+        [[highlight]]setState(() {[[/highlight]]
+          [[highlight]]// the state that has changed here is the animation object’s value[[/highlight]]
+        [[highlight]]});[[/highlight]]
+      [[highlight]]});[[/highlight]]
+    [[highlight]]controller.forward();[[/highlight]]
+  [[highlight]]}[[/highlight]]
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: new EdgeInsets.symmetric(vertical: 10.0),
+        height: [[highlight]]animation.value,[[/highlight]]
+        width: [[highlight]]animation.value,[[/highlight]]
+        child: new FlutterLogo,
+      ),
+    );
+  }
+}
+
+void main() {
+  runApp(new LogoApp());
+}
+{% endprettify %}
+
+The `addListener()` function calls `setState()`, so every time the
+Animation generates a new number, the current frame is marked dirty,
+which forces `build()` to be called again.
+In `build()`, the container changes size because its height and width
+now use `animation.value` instead of a hardcoded value.
+
+With these few changes, you’ve created your first animation in Flutter!
+You can find the source for this example,
+[animate1,](https://raw.githubusercontent.com/InMatrix/animation_tutorial/master/lib/animate1.dart)
+on GitHub.
+
+<aside class="alert alert-success" markdown="1">
+**Dart language tricks**
+You may not be familiar with Dart's cascade notation&mdash;the two
+dots in `..addListener()`. This syntax means that the `addListener()`
+method is called with the return value from `animate()`.
+Consider the following example:
+
+<!-- skip -->
+{% prettify dart %}
+[[highlight]]animation = tween.animate(controller)[[/highlight]]
+          [[highlight]]..addListener(()[[/highlight]] {
+            setState(() {
+              // the animation object’s value is the changed state
+            });
+          });
+{% endprettify %}
+
+This code is equivalent to:
+
+<!-- skip -->
+{% prettify dart %}
+[[highlight]]animation = tween.animate(controller);[[/highlight]]
+[[highlight]]animation.addListener(()[[/highlight]] {
+            setState(() {
+              // the animation object’s value is the changed state
+            });
+          });
+{% endprettify %}
+
+The difference is that, in the first case,
+the animation variable hasn't been set when the closure is created.
```

Comment:
that doesn't matter. What matters is whether the variable is in scope, and it is, so both techniques would work fine.